### DISCUSSION TITLE
Prisma sometimes creates long relation names by appending table and column names together

### DISCUSSION THREAD

**Discussion by tomashaddad** [prisma/prisma#24427] [Q&A] [answered]
### Question

When introspecting our database with `npx prisma db pull` sometimes Prisma appends the field used to define the foreign key relation to the related field in the other table. Take these models:

```prisma
model variation_requests {
  id             Int      @id @default(autoincrement())
  codeid         Int
  ...
  jobcodes       jobcodes @relation(fields: [codeid], references: [codeid], onDelete: NoAction, onUpdate: NoAction)
}

model variation_submissions {
  id                                                   Int       @id @default(autoincrement())
  created_by                                           Int?
  reviewed_by                                          Int?
  ...
  accounts_accountsTovariation_submissions_created_by  accounts? @relation("accountsTovariation_submissions_created_by", fields: [created_by], references: [username_id], onDelete: NoAction, onUpdate: NoAction)
  accounts_accountsTovariation_submissions_reviewed_by accounts? @relation("accountsTovariation_submissions_reviewed_by", fields: [reviewed_by], references: [username_id], onDelete: NoAction, onUpdate: NoAction)
}
```

Notice how `variation_requests` contains just `jobcodes`, so connecting `codeid` to an existing jobcode is simple during `create` using: `prisma_client.variation_requests.create({ ..., jobcodes: { connect: { codeid } } })`.

However if I want to connect to an existing user upon creating a variation_submission I need to reference `accounts_accountsTovariation_submissions_created_by`. Why doesn't Prisma generate it simply as `accounts` upon introspection?

### Expected behavior (optional)

I expect the connecting relation to simply be named the referred table.

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: Windows
- Database: Postgresql
- Node.js: v20.13.1
- Prisma: 4.6.1

**Comment by janpio** [comment, marked as answer]
This is to differentiate the two relations to `accounts` clearly. Otherwise we would need to make a decision which one is to be called `accounts` and which one gets the long name - and in ~50% of cases the user would probably disagree with that choice :D 

You can just rename these to whatever you want, and future Introspections should recognize that and keep the name around.

**Reply by tomashaddad** [reply]
No idea how I didn't realise that myself ... 😭 Thanks for the tip at the end.

**Reply by tomashaddad** [reply]
Sorry @janpio had a think about this overnight and I have another quick question, why not just prepend the other table to the column name? So in the above it would be `accounts_created_by` and `accounts_reviewed_by`.

**Reply by janpio** [reply]
Very good question. What does the other side of the relation look like after Introspection?

I would guess it is just the _safer_ option. We fully expect these relation field names to be adapted (`F2` with our VS Code extension installed should make it easy to update all instances in the Schema file at the same time, and type errors should highlight and usages in your application code) so did probably not prioritize conciseness.

But if there is something wrong and could be simplified, no reason we can't still do that.
